### ISSUE TITLE
Texture modifiers and imageCleanTransparent are too slow

### ISSUE THREAD

**Issue by Desour** [minetest/minetest#14322]
Performance issue.

This issue appears in at least two places:
* Joining a server takes longer. (Related: #11829)
* Loading objects with new textures (e.g. signs) freezes the client for a short time.

## How to test

* Enable any texture filter stuff, like mip mapping, to trigger `[applyfiltersformesh`/`imageCleanTransparent` stuff.
* Install `signs_lib` (<https://github.com/mt-mods/signs_lib>) and some luacmd fork (<https://github.com/Desour/luacmd>) in a mtg world.
* Create some signs with different text each: `/lua local heren = here:round() for x = 0, 8 do for y = 0, 8 do for z = 0, 8 do local p = heren:offset(x,y,z) core.set_node(p, {name="default:sign_wall_wood",param2=2}) core.get_meta(p):set_string("text", p:to_string()..string.rep("\nAAAAAAA",5)) end end end`
* Rejoin.
* Notice how the client freezes. when the sign objects load.

Here's a flamegraph:

![2024-01-29-220615_1920x1080_scrot](https://github.com/minetest/minetest/assets/7613443/cc51b25a-e929-41e6-a850-cf81aaeeb45b)

**Comment by sfan5** [comment]
Is this still the case? I don't think so.
Other filters are still "unaccelerated" but that doesn't seem a major issue.

**Comment by Desour** [comment]
>Is this still the case?

Yes. It got better, but imageCleanTransparent at least is still slower than it has to be (i.e. could be improved with #14478, and/or by moving it to the gpu). And while it might not be the worst, it still takes a significant part in join time. And I can still reproduce the sign issue (not only with the extreme case given here, but I can also run around a server spawn and experience small freezes).

**Comment by sfan5** [comment]
#17372 btw.

In case of signs maybe we could introduce some way to delay-load entity textures?
